### PR TITLE
Re write routing to query based system

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- no visual changes
- routing is switched to query based system
  - instead of insane nesting and params ambiguity, every entity id is clearly defined by query param in URL
- created new consistent breadcrumbs system
  - breadcrumbs can be generated from URL pathname
  - every root page controls when to update it

### CODE REVIEW

File: src/smart-components/portfolio/portfolio-item-detail/detail-toolbar-actions.js [hunk 1]
```diff
@@ -20,25 +20,18 @@ const DetailToolbarActions = ({
   setOpen,
   isFetching
 }) => {
```
Comment (rvsia) [inline src/smart-components/portfolio/portfolio-item-detail/detail-toolbar-actions.js]: unnecessary block

**Comment by rvsia** [comment]
@Hyperkid123 Breadcrumbs do not work properly

badbreadcrumbs

File: src/helpers/shared/breadcrumbs-creators.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+export const FRAGMENT_TITLE = {
+  '/portfolio': (getState) =>
+    getState().portfolioReducer.selectedPortfolio.name,
+  '/portfolio/portfolio-item': (getState) =>
+    getState().portfolioReducer.portfolioItem.portfolioItem.name,
+  '/portfolio/portfolio-item/edit-survey': () => 'Edit survey',
+  '/portfolio/portfolio-item/add-products': () => 'Add products',
+  '/platform': (getState) => getState().platformReducer.selectedPlatform.name,
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply src/helpers/shared/breadcrumbs-creators.js]: Nope

File: src/presentational-components/platform/platform-item.js [hunk 3]
```diff
@@ -5,17 +5,12 @@ import { Card, CardHeader, CardFooter } from '@patternfly/react-core';
...
-const PlatformItem = (props) => (
+const PlatformItem = ({ src, ...props }) => (
   <Card key={props.id} className="content-gallery-card">
     <CardHeader>
-      <CardIcon
-        src={`${TOPOLOGICAL_INVENTORY_API_BASE}/service_offering_icons/${props.service_offering_icon_id}/icon_data`}
-        style={{ height: 40 }}
-        sourceId={props.source_id}
-      />
+      <CardIcon src={src} style={{ height: 40 }} />
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply src/presentational-components/platform/platform-item.js]: In future

File: src/presentational-components/shared/top-toolbar.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ import {
 } from '@patternfly/react-core';
 import clsx from 'clsx';
 import { ToolbarTitlePlaceholder } from './loader-placeholders';
-import CatalogBreadcrumbs from './breadcrubms';
+import CatalogBreadcrumbs from '../../smart-components/common/catalog-bread-crumbs';
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply src/presentational-components/shared/top-toolbar.js]: lol

**Comment by Hyperkid123** [comment]
@rvsia there was a missing default trigger for breadcrumbs re-render. Should be fixed soon.

File: src/router.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+let release = '/';
+if (pathName[0] === 'beta') {
+  release = `/${pathName.shift()}/`;
+}
+
+const Router = () => {
+  return (
+    <AppContext.Provider value={{ release }}>
+      <BrowserRouter basename={`${release}${pathName[0]}/${pathName[1]}`}>
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply src/router.js]: I need the release and I cant do that with the function from front-end components

**Comment by rvsia** [comment]
@Hyperkid123 Tabs breadcrumbs do not work properly

breadcrumbs2

**Comment by Hyperkid123** [comment]
@rvsia I think we should remove the breadcrumbs from tabs since it is not a "full" route

**Comment by rvsia** [comment]
@Hyperkid123 Yeah, I totally agree. :+1:

**Comment by Hyperkid123** [comment]
@rvsia ok so I moved the strings to constants and removed the breadcrumbs from the tabs. Also fixed some code smells. It should be OK now but I would appreciate if you could click through the app again.

**Comment by rvsia** [comment]
@Hyperkid123 

Yes, I will check it tomorrow. And don’t worry about the size of this PR, last time I was working on breadcrumbs, it took only 8 months.. 😸

**Comment by lgalis** [comment]
@Hyperkid  - Copy portfolio also fails

**Comment by lgalis** [comment]
@Hyperkid123 - copy portfolio also fails

File: src/smart-components/common/catalog-breadcrumbs.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <Breadcrumb>
+      {fragments.map(({ title, pathname, searchParams }, index) => (
+        <ConditionalLink
+          pathname={!(fragments.length === index + 1) ? pathname : undefined}
+          searchParams={searchParams}
+          nav
+          exact
+          key={pathname}
+          className="pf-c-breadcrumb__item"
```
Comment (rvsia) [inline src/smart-components/common/catalog-breadcrumbs.js]: You should check here if it is active, otherwise you dont want to add `cursor: pointer` to it

Comment (Hyperkid123) [reply src/smart-components/common/catalog-breadcrumbs.js]: Fixed that with CSS.

**Comment by Hyperkid123** [comment]
@lgalis I fixed the issue with routing. @rvsia styling was fixed.

**Comment by Hyperkid123** [comment]
Share/unshare notifications are missing

**Comment by rvsia** [comment]
@Hyperkid123 
addproducts

**Comment by Hyperkid123** [comment]
@rvsia fixed

**Pull request merged**